lib/driver.js

```
@@ -484,6 +484,33 @@ export class XCUITestDriver extends BaseDriver {
     }
   }
 
+  async startMjpegStream() {
+      // turn on mjpeg stream reading if requested
+      if (this.opts.mjpegScreenshotUrl) {
+        this.log.info(`Starting MJPEG stream reading URL: '${this.opts.mjpegScreenshotUrl}'`);
+        this.mjpegStream = new mjpeg.MJpegStream(this.opts.mjpegScreenshotUrl);
+        await this.mjpegStream.start();
+      }
+  }
+
+  async allocateMjpegServerPort(caps) {
+    this.log.debug(
+      `MJPEG broadcasting requested, forwarding MJPEG server port ${this.opts.mjpegServerPort} ` +
+      `to local port ${this.opts.mjpegServerPort}`
+    );
+    try {
+      await DEVICE_CONNECTIONS_FACTORY.requestConnection(caps.udid, this.opts.mjpegServerPort, {
+        devicePort: this.opts.mjpegServerPort,
+        usePortForwarding: true,
+      });
+    } catch {
+       throw new Error(
```

Please do not silence possible exceptions

---

@mykola-mokhnach We are not silencing the error, we are actually throwing in catch block. Can you pls clarify

---

The question here is whether we would like to bubble up actual exceptions happening inside the factory or we would like to customize them by providing the particular error text

---

@mykola-mokhnach I just kept it as per our screenrecord.js file.

---

Makes sense. I just wanted to make the error more helpful for end users, so they could address easily it in case of failure. 

How about the following:

- We print the error stack using `log.debug(e.stack)`
- We show the following error message: `Cannot ensure MJPEG broadcast functionality by forwarding the local port XXX requested by the 'YYY' capability to the device port XXX. Original error: ${e.message}`